src/index.rs

```diff
@@ -0,0 +1,33 @@
+use std::io;
+use crate::metadata::get_ids;
+
+pub struct Index {
+    pub packets: Vec<Packet>
+}
+
+pub struct Packet {
```

Comment:
I think we need it to be public so that we can loop over it in query.rs. Otherwise getting error like
```
error: type `Packet` is private
  --> src/query.rs:67:8
   |
67 |     Ok(index.packets.last().unwrap().id.clone())
   |        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ private type
```

Though yeah I think you're right, I'm not sure having a `Packet` object is helping at moment. Probably some more efficient way to represent that in the index. Will have a think

Reply:
Ok having a look around I think this is probably a good way to start trying to encapsulate the filtering logic into one place. I think probably I'd like to leave this as is for now and if we see the querying is slow we can look at some other approaches.